waveorder/io/reader.py

```diff
@@ -0,0 +1,158 @@
+import numpy as np
+import os
+import zarr
+from tifffile import TiffFile
+
+
+class MicromanagerReader:
+
+    def __init__(self, folder):
+        """
+        loads ome-tiff files in folder into zarr or numpy arrays
+        Strategy:
+            1. read all files in folder
+            2. search for the file whose omexml does not contain the element tag "BinaryOnly"-> this is the master ome-tiff
+            3. each series represents a micro-manager stage position.  upon call for data, assign this data to a dict
+
+        :param folder: folder containing all ome-tiff files
+        :param reader: tifffile or aicsimageio
+        """
+
+        self._positions = {}
+        self.mm_meta = None
+        self.stage_positions = 0
+        self.height = 0
+        self.width = 0
+        self.frames = 0
+        self.slices = 0
+        self.channels = 0
+        self.chnames = []
+
+        self.master_ome_tiff = self._get_master_ome_tiff(folder)
+        print(self.master_ome_tiff)
+        self._set_mm_meta()
+
+    def _set_mm_meta(self):
+        with TiffFile(self.master_ome_tiff) as tif:
+            self.mm_meta = tif.micromanager_metadata
+
+            if self.mm_meta['Summary']['Positions'] > 1:
+                self.stage_positions = []
+                for p in range(self.mm_meta['Summary']['Positions']):
+                    print(f"appending {self.mm_meta['Summary']['StagePositions'][p]}")
+                    self.stage_positions.append(self.mm_meta['Summary']['StagePositions'][p])
+
+            for ch in self.mm_meta['Summary']['ChNames']:
+                print(f"appending {ch}")
+                self.chnames.append(ch)
+
+            self.height = self.mm_meta['Summary']['Height']
+            self.width = self.mm_meta['Summary']['Width']
+            self.frames = self.mm_meta['Summary']['Frames']
+            self.slices = self.mm_meta['Summary']['Slices']
+            self.channels = self.mm_meta['Summary']['Channels']
+
+    def _extract_data(self, master_ome):
+        """
+        extract all series from ome-tiff and place into dict of (pos: zarr)
+        :param master_ome: full path to master OME-tiff
+        :return:
+        """
+        with TiffFile(master_ome) as tif:
+            for idx, tiffpageseries in enumerate(tif.series):
+                self._positions[idx] = zarr.open(tiffpageseries.aszarr(), mode='r')
+
+    def _get_master_ome_tiff(self, folder_):
+        """
```

Comment:
Using the ome.tiff datasets you provided, I got a warning like:
![image](https://user-images.githubusercontent.com/20444608/107723941-7d1f5900-6c97-11eb-8638-bbf3c5128a20.png)

What does this mean and do we need to pay attention to it?

Reply:
Sadly this is a result of tifffile library.  I'd have to do some digging to find out what is invalid about the micro-manager JSON.  There must be some formatting standard that micro-manager does not adhere to.

Reply:
I recently discussed OME-TIFFs and OME-zarr's with Nico. He pointed out that the data formats that work the best for image acquisition (where you want to write images as they come) and image analysis (where you want to structure images such that they are easy to parse) is different. For this reason, Micro-Manager OME-TIFFs are not fully compatible with OME-TIFF standard. Does this error affect the reading of planes? If not, it is safe to ignore.

Reply:
Interesting point and definitely worth keeping in mind as we work with Allen Inst.
For now i'm implementing a logger that is the same as what Aicsimageio folks use.  You can set the reporting level to `ERROR` or `CRITICAL` which will suppress the JSON and other warnings even from tifffile library.